lib/ruby_smb/client/authentication.rb

```diff
@@ -201,6 +201,9 @@ def smb1_type2_message(response_packet)
       def smb2_authenticate
         response = smb2_ntlmssp_negotiate
         challenge_packet = smb2_ntlmssp_challenge_packet(response)
+        if @dialect == '0x0311'
```

Comment:
For now, `0x0311` dialect is the only one that needs a pre-auth integrity hash to generate the private keys for encryption/decryption. If a new dialect version comes out, we will need to update the client to add it to the negotiation logic to be supported and will add it here if needed. Also, since Microsoft often changes the logic when a new version comes out we will probably have to redo all of this anyway :).